java/maven.hints/src/org/netbeans/modules/maven/hints/pom/UseReleaseOptionHint.java

```diff
@@ -0,0 +1,241 @@
+
+    @Override
+    public List<ErrorDescription> getErrorsForDocument(POMModel model, Project prj) {
+
+        List<ErrorDescription> hints = new ArrayList<>();
+
+        Build build = model.getProject().getBuild();
+        if (build != null) {
+            for (Plugin plugin : build.getPlugins()) {
+                if ("maven-compiler-plugin".equals(plugin.getArtifactId())) {
+                    if (!isPluginCompatible(plugin)) {
+                        return Collections.emptyList();
+                    }
+                    if (plugin.getConfiguration() != null) {
+                        hints.addAll(createHintsForParent("", plugin.getConfiguration()));
+                    }
+                    if (plugin.getExecutions() != null) {
+                        for (PluginExecution execution : plugin.getExecutions()) {
+                            if (execution.getConfiguration() != null) {
+                                hints.addAll(createHintsForParent("", execution.getConfiguration()));
+                            }
+                        }
+                    }
+                    break;
+                }
+            }
+        }
+
+        Properties properties = model.getProject().getProperties();
+        if (properties != null) {
+            hints.addAll(createHintsForParent("maven.compiler.", properties));
+        }
+
+        return hints;
+    }
+
+    private List<ErrorDescription> createHintsForParent(String prefix, POMComponent parent) {
+
+        int source;
+        int target;
+
+        // property name or an int value
+        String release = null;
+
+        try {
+            String sourceText = parent.getChildElementText(POMQName.createQName(prefix+SOURCE_TAG, true));
+            if (isProperty(sourceText)) {
+                release = sourceText;
+                sourceText = getProperty(sourceText, parent.getModel());
+            }
+
+            String targetText = parent.getChildElementText(POMQName.createQName(prefix+TARGET_TAG, true));
+            if (isProperty(targetText)) {
+                release = targetText;
+                targetText = getProperty(targetText, parent.getModel());
+            }
+
+            source = Integer.parseInt(sourceText);
+            target = Integer.parseInt(targetText);
+            if (release == null) {
+                release = String.valueOf(target);
+            }
+        } catch (NumberFormatException ignored) {
+            return Collections.emptyList();
+        }
+
+        if (source == target && source >= 9) {
+            List<ErrorDescription> hints = new ArrayList<>();
+            for (POMComponent prop : parent.getChildren()) {
+                String name = prop.getPeer().getNodeName();
+                if (name.equals(prefix+SOURCE_TAG) || name.equals(prefix+TARGET_TAG)) {
+                    hints.add(createHintForComponent(prefix, prop, parent.getModel(), release));
```

Comment:
release replaces source/target in javac. You can't mix them up:
```
javac -source 11 --release 11
error: option --source cannot be used together with --release
```
So I don't check if there is a release already there (the javac config would be already failing). It would ignore it right now and add another release element to the xml.

Reply:
the only reason to compute release is to do the `>= 9` check. Otherwise it would simply collapse source and target to release if they are equal strings or properties which point to equal strings

Reply:
Yes, javac will error when release + source/target is present. But please add the check since users are creative, and may make unintentionally an erroneous pom (source, target AND release). A hint+fix that appears, removes source+target and **duplicates** the release will add even more confusion.

Reply:
pick your favorite:
1) should it remove the old release tag if it finds one (and add the new one later)
2) it only removes source/target but doesn't add another release tag if there is one (even if it has the wrong value)

Reply:
implementing 1 for now since its fairly elegant to implement